tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs

```diff
@@ -31,130 +30,107 @@ public void Constructor_NullRepository_Throws()
-
-            var ordersResult = objectResult.Value as List<OrderModel>;
-            ordersResult.Count().Should().Be(0);
+            var result = await controller.GetAllAsync(Guid.Empty) as OkObjectResult;
+            var orders = result.Value as List<OrderModel>;
+            orders.Count.Should().Be(0);
         }
 
         [TestCase(null, null)]
         [TestCase("C0000014-01", "Some Description")]
         [TestCase("C0000014-02", null)]
-        public async Task GetAllAsync_SingleOrderExists_ReturnsTheOrder(string orderId, string orderDescription)
+        public async Task GetAllAsync_SingleOrderWithOrganisationIdExists_ReturnsTheOrder(string orderId, string orderDescription)
         {
-            var model = new List<Order>()
-            {
-                new Order()
-                {
-                    OrderId = orderId,
-                    Description = orderDescription,
-                    LastUpdatedBy = Guid.NewGuid(),
-                    LastUpdated = DateTime.UtcNow,
-                    Created = DateTime.UtcNow,
-                    OrderStatus = submittedStatus
-                }
-            };
+            var organisationId = Guid.NewGuid();
 
-            var expected = new List<OrderModel>()
+            var orders = new List<(Order order, OrderModel expected)>
             {
-                new OrderModel()
-                {
-                    OrderId = model[0].OrderId,
-                    OrderDescription = model[0].Description,
-                    LastUpdatedBy = model[0].LastUpdatedBy,
-                    LastUpdated = model[0].LastUpdated,
-                    DateCreated = model[0].Created,
-                    Status = model[0].OrderStatus.Name
-                }
+                CreateOrderTestData(orderId, organisationId, orderDescription)
             };
 
+            var context = OrdersControllerTestContext.Setup();
+            context.Orders = orders.Select(x => x.order);
+
+            using var controller = context.OrdersController;
+
+            var result = await controller.GetAllAsync(organisationId) as OkObjectResult;
+            var ordersResult = result.Value as List<OrderModel>;
+            ordersResult.Count.Should().Be(1);
+            ordersResult.Should().BeEquivalentTo(orders.Select(x => x.expected));
+        }
 
-            using var controller = OrdersControllerBuilder.Create().WithListOrders(model).Build();
+        [Test]
+        public async Task GetAllAsync_MultipleOrdersWithOrganisationIdExist_ReturnsAllOrders()
+        {
+            var organisationId = Guid.NewGuid();
 
-            var result = await controller.GetAllAsync();
+            var orders = new List<(Order order, OrderModel expected)>
+            {
+                CreateOrderTestData("C0000014-01", organisationId, "Some Description"),
+                CreateOrderTestData("C000012-01", organisationId, "Another Description")

```

Comment:
Up to personal preference, but have you considered getting this data from the `[TestCase]` annotation (the same way as you're doing 1 test up)?

Reply:
Potential for a `TestCaseSource`. See `ApplicationUserValidatorTests`. Though I am not suggesting change it at this stage.